### PR TITLE
feat: sort active medications to top of eChart medication list

### PR DESCRIPTION
**Pull request by yingbull**
### **User description**
Port of openo-beta/open-o#2307 by LiamStanziani.

Active medications (current and not archived, or long-term) now display
above expired/inactive ones in the eChart Medications tab. This improves
clinical usability by letting providers quickly find current prescriptions
without scrolling through expired entries.

Changes:
- Add ACTIVE_FIRST comparator with stable sort on uniqueDrugs list
- Extract isActiveDrug() helper shared by sort and getClassColour()
- Add 8 unit tests covering sort order, stability, and edge cases

https://claude.ai/code/session_011sjsotiSK6vrW5vHX2zf5U

___

### **Description**
- Introduced a new comparator `ACTIVE_FIRST` to sort active medications above expired ones in the eChart.
- Added a helper method `isActiveDrug()` to determine if a medication is active.
- Implemented unit tests to validate the sorting logic and ensure stability of order.
- Improved clinical usability by allowing providers to quickly identify current prescriptions.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>EctDisplayRx2Action.java</strong><dd><code>Enhance medication sorting logic in EctDisplayRx2Action</code>&nbsp; &nbsp; </dd></summary>
<hr>

src/main/java/io/github/carlos_emr/carlos/encounter/pageUtil/EctDisplayRx2Action.java
<li>Added <code>ACTIVE_FIRST</code> comparator for sorting medications.<br> <li> Implemented <code>isActiveDrug()</code> helper method for determining active <br>status.<br> <li> Updated medication sorting logic to prioritize active medications.<br>

</details>

  </td>
  <td><a href="https://github.com/carlos-emr/carlos/pull/500/files#diff-8ef90d310c71d3b847a1769c665f546a142d2e7d8b6bfbe446993264b37f7c09">+30/-1</a>&nbsp; &nbsp; </td>
</tr>
</table></td></tr><tr><td><strong>Tests
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>EctDisplayRx2ActionSortTest.java</strong><dd><code>Add unit tests for medication sorting logic</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/test-modern/java/io/github/carlos_emr/carlos/encounter/pageUtil/EctDisplayRx2ActionSortTest.java
<li>Added unit tests for medication sorting functionality.<br> <li> Verified sorting behavior for active, expired, and long-term <br>medications.<br> <li> Ensured stability of sort order within medication groups.<br>

</details>

  </td>
  <td><a href="https://github.com/carlos-emr/carlos/pull/500/files#diff-a06238d0e51b145b74348f1c9536c207456fc84e6ad5b8c62226cf2284be87c6">+170/-0</a>&nbsp; </td>
</tr>
</table></td></tr></tr></tbody></table>

___

> 💡 **Penify usage**:
>Comment `/help` on the PR to get a list of all available Penify tools and their descriptions

## Summary by CodeRabbit

* **New Features**
  * Prescriptions are now automatically sorted with active medications displayed first in the encounter view. This includes current non-archived treatments and long-term prescriptions, with expired and archived medications listed below, enhancing visibility of current patient treatments for faster clinical review.

### CODE REVIEW

File: src/main/java/io/github/carlos_emr/carlos/encounter/pageUtil/EctDisplayRx2Action.java [hunk 1]
```diff
@@ -79,6 +83,11 @@ public boolean getInfo(EctSessionBean bean, HttpServletRequest request, NavBarDi
             CppPreferencesUIBean prefsBean = new CppPreferencesUIBean(loggedInInfo.getLoggedInProviderNo());
             prefsBean.loadValues();
 
+            // Sort active medications to the top of the list, preserving
+            // relative order within each group (stable sort).
+            // Lower value = sorted higher in the list (0 = active first, 1 = non-active after).
+            uniqueDrugs.sort(ACTIVE_FIRST);
+
```
[the remark this answers is not in the document]
Comment (yingbull) [reply src/main/java/io/github/carlos_emr/carlos/encounter/pageUtil/EctDisplayRx2Action.java:90]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/carlos-emr/carlos/pull/500#discussion_r2865756812)
